src/main/java/org/util/string/StringUtil.java

```diff
@@ -1,9 +1,735 @@
+    }
+
+    @Override
+    public String toString() {
+        return builder.toString();
+    }
+
+    public StringUtil soundex() {
+
+        char[] ch = builder.toString().toUpperCase().toCharArray();
+
+        for (int i = 1; i < 4; i++) {
+
+            switch (ch[i]) {
+                case 'B':
+                case 'P':
+                case 'F':
+                case 'V':
+                    ch[i] = '1';
+                    break;
+                case 'C':
+                case 'S':
+                case 'K':
+                case 'G':
+                case 'J':
+                case 'Q':
+                case 'X':
+                case 'Z':
+                    ch[i] = '2';
+                    break;
+                case 'D':
+                case 'T':
+                    ch[i] = '3';
+                    break;
+                case 'L':
+                    ch[i] = '4';
+                    break;
+                case 'M':
+                case 'N':
+                    ch[i] = '5';
+                    break;
+                case 'R':
+                    ch[i] = '6';
+                    break;
+
+                default:
+                    ch[i] = '0';
+                    break;
+
+
+            }
+        }
+        char[] ch1 = new char[4];
+        for (int i = 0; i < 4; i++) {
+            ch1[i] = ch[i];
+        }
+
+
+        return new StringUtil(new String(ch1));
+    }
+
+    @Override
+    public int compareTo(StringUtil o) {
+
+        if (builder.equals(o))
+            return 0;
+        else if (builder.length() - o.toString().length() > 0)
+            return 1;
+        else
+            return -1;
+    }
+
+    public byte[] bytes() {
+        return builder.toString().getBytes();
+    }
+
+
+    public byte[] bytes(Charset charset) {
+        return builder.toString().getBytes(charset);
+    }
+
+    private static class BuilderIterator implements Iterator<Character> {
+
+        private final StringBuilder builder;
+        private final int cursor;
+        int current = 0;
+
+
+        public BuilderIterator(StringBuilder builder) {
+            this.builder = builder;
+            this.cursor = 0;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return current < builder.capacity();
+        }
+
+        @Override
+        public Character next() {
+            char[] ch = builder.toString().toCharArray();
+
+            if (builder.capacity() > current)
+                return ch[current++];
+            else
+                throw new NoSuchElementException();
+        }
+    }
+
+    @Override
+    public Iterator<Character> iterator() {
+        return new BuilderIterator(builder);
+    }
+
+    public StringUtil remove(int from, int to) {
+        return new StringUtil(builder.delete(from, to));
+    }
+
+    // returns amount of occurences
+    public int find(char value) {
+        int count = 0;
+        for (char ch : builder.toString().toCharArray()) {
```

Comment:
you don't need convert build to string, you don't need convert string to char array to find anything